icepyx/core/APIformatting.py

```diff
@@ -282,31 +283,14 @@ def __init__(
         self._fmted_keys = values if values is not None else {}
 
     @property
-    def poss_keys(self):
+    def poss_keys(self) -> dict[str, list[str]]:
```

Comment:
This property returns static data, from what I can tell, so I simplified it down, and removed `_poss_keys` attribute.

Reply:
I think I had it this way because (1) I'd learned about properties and was probably overusing them; (2) I wanted it to be private enough the user understood they could look but shouldn't touch (so an invalid parameter wasn't submitted). Overall a fan of the simplification though.

Reply:
>(1) I'd learned about properties and was probably overusing them;

Who can blame you, properties are awesome!

>(2) I wanted it to be private enough the user understood they could look but shouldn't touch

Do you mean "can't set it" by "shouldn't touch"?

Reply:
> Do you mean "can't set it" by "shouldn't touch"?

I suppose so.

Reply:
Gotcha. That's already covered by the `@property` decorator :)

```python
>>> class Foo:
...   @property
...   def foo(self):
...     return "foo"
... 
>>> Foo().foo
'foo'
>>> Foo().foo = "bar"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't set attribute 'foo'
```